### PR TITLE
Initial commit for Prime_E w/ VIA config support

### PR DESCRIPTION
**Pull request by holtenc**
## Description

Initial commit for Prime_E with support for VIA configurator. Additionally added VIA configurator support to /handwired/prime_exl for personal use. 

Discussed this addition with @Wilba6582 and he has appoved it. (https://github.com/olivia/via-config/pull/52)

Honestly not sure why the files in lib/lufa and ...projects are part of this PR. I've not made any changes to them. Is there a way to exclude them?

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/lufa/Projects/LEDNotifier/CPUUsageApp/obj/Debug/CPUMonitor.csproj.CoreCompileInputs.cache [hunk 1]
```diff
@@ -0,0 +1 @@
+7e58a720d521047f73ccc9b7cabe3901380feb6d
```
Comment (drashna) [inline lib/lufa/Projects/LEDNotifier/CPUUsageApp/obj/Debug/CPUMonitor.csproj.CoreCompileInputs.cache]: Could this be reverted? 

You can run: 
```bash
git checkout -- lib/
```
That should fix all of these.

File: keyboards/primekb/prime_e/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    LAYOUT(
+		KC_ESC,   KC_Q,     KC_W,     KC_E,   KC_R,    KC_T,          KC_Y,   KC_U,     KC_I, KC_O,    KC_P,    KC_DEL, KC_BSPC,
+		KC_TAB,   KC_A,     KC_S,     KC_D,   KC_F,    KC_G,          KC_H,   KC_J,     KC_K, KC_L,    KC_QUOT, KC_ENT,
+		KC_LSFT,  KC_Z,     KC_X,     KC_C,   KC_V,    KC_B,          MO(1),  KC_N,     KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT,
+		KC_LCTL,  KC_LALT,                    KC_RBRC, MO(1),	        KC_SPC, KC_MINS,                          KC_LGUI, TO(1)
+    )
+};
+
+void matrix_init_user(void) {
```
Comment (drashna) [inline keyboards/primekb/prime_e/keymaps/via/keymap.c]: Same as above.

**Comment by holtenc** [comment]
I think that should resolve all your requested changes.

**Comment by drashna** [comment]
Looks good.  Though, the lib stuff is still there, unfortunately. 

I'm not 100% sure how to fix that (I'm not git expert, either).... but I think you can do `git reset upstream/master lib` to remove those changes.  `--hard` will remove them completely, but since I'm not 100% sure, so without should be safer.

**Comment by holtenc** [comment]
OK. got those Lib files removed. Sorry.

**Comment by drashna** [comment]
Awesome! Thanks! 

And no worries. Stuff like that happens from time to time.  Git is some serious dark magic sometimes. :)

**Comment by drashna** [comment]
And @Wilba6582, just wanted to get your okay before merging this.

**Comment by Wilba6582** [comment]
@drashna yes, this was in the works before the cutoff

**Comment by drashna** [comment]
Awesome, just wanted to make sure!

**Pull request merged**